ui/lib/core/addon/templates/components/replication-action-disable.hbs

```diff
@@ -1,67 +1,45 @@
-      Since the cluster is currently bootstrapping, we need to know which mode to disable.
-      Be sure to choose it below.
-      <label for="replication-mode" class="is-label">
-        Replication cluster mode
-      </label>
-      <div class="field is-expanded">
-        <div class="control select is-fullwidth">
-          <select onchange={{action (mut mode) value="target.value"}} id="replication-mode" name="replication-mode">
-            {{#each (array 'primary' 'secondary') as |modeOption|}}
-              <option
-                selected={{if mode (eq mode modeOption) (eq modeOption 'primary')}}
-                value={{modeOption}}
-              >
-                {{modeOption}}
-              </option>
-            {{/each}}
-          </select>
-        </div>
-      </div>
-    {{else}}
-      The cluster will no longer be able to connect to the primary.
-    {{/if}}
-    <AlertInline
-      @type="danger"
-      @message="Caution: re-enabling this node as a primary or secondary will change its cluster ID."
-    />
-  </p>
-  <p>
-    In the secondary case this means a wipe of the
-    underlying storage when connected to a primary, and in the primary case,
-    secondaries connecting back to the cluster (even if they have connected
-    before) will require a wipe of the underlying storage.
-  </p>
-</div>
-<div class="field">
-  <div class="control">
-    <ConfirmAction
-      @buttonClasses="button is-primary"
-      @confirmTitle="Disable this cluster?"
-      @confirmMessage="Data in this cluster will no longer be replicated."
-      @confirmButtonText="Disable"
-      @horizontalPosition="auto-left"
-      @onConfirmAction={{action
-        "onSubmit"
-        "disable"
-        (if
-          (eq model.replicationAttrs.modeForUrl 'bootstrapping')
-          mode
-          model.replicationAttrs.modeForUrl
-        )
-      }}
-      data-test-disable-replication="true"
-    >
+<div class="action-block is-rounded">
+  <div class="action-block-info">
+    <p class="action-block-title">Disable Replication</p>
+    <p>
+      Disable {{replicationDisplayMode}} Replication entirely on the cluster.
+    </p>
+  </div>
+
+  <div class="action-block-action">
+    <button
+      class="button is-danger"
+      onclick={{action (mut isModalActive) true}}>
       Disable Replication
-    </ConfirmAction>
+    </button>
   </div>
 </div>
+
+<ConfirmationModal
+  @title="Disable Replication?"
+  @onClose={{action (mut isModalActive) false}}
+  @isActive={{isModalActive}}
+  @confirmText={{if (eq replicationDisplayMode "DR") "Disaster Recovery" replicationDisplayMode}}
+  @toConfirm="disabling {{replicationDisplayMode}} replication on this cluster"
+  @onConfirm={{action
+    "onSubmit"
+    "disable"
+    (if
+      (eq model.replicationAttrs.modeForUrl 'bootstrapping')
+      mode
+      model.replicationAttrs.modeForUrl
+    )
+  }}
+>
+  <p>
+    Disabling Performance replication entirely on this secondary cluster means that:
```

Comment:
looks like the content here needs to be swapped out depending on whether it's a primary or secondary -- which you should have access to on the `model`, or perhaps from this `clusterMode` property? https://github.com/hashicorp/vault/blob/5a75ef2f7aae44b7a94bdd96738dc074efbef955/ui/app/templates/vault/cluster/replication-dr-promote/index.hbs#L34

looks like that property needs to be added to the list of properties inside the `ReplicationActions` component too: https://github.com/hashicorp/vault/blob/f60d159674d64aa8b58ca3bcf51f6d54d4eacef3/ui/lib/core/addon/components/replication-actions.js#L20

Reply:
Great catch, I've updated the content per Ivana's mockups!

Reply:
I was actually able to use the values that were already here, but I'd love a sanity check on the logic: 

- Only DR Secondaries would need to pass `clusterMode` explicitly
- Everywhere else `replicationAttrs` is available on the model, which has an `isPrimary` value

SO my assumption is that if `isPrimary` is not true, the node is a secondary.  Or, we can do that logic from the `replication-actions` file and pass it in explicitly 🤔

Reply:
regarding #1, yeah, it's strange that DR secondaries seem to be the only place where you have to explicitly pass it in. could we refactor it so you don't have to? that way you could get rid of the conditional here: https://github.com/hashicorp/vault/pull/9061/files#diff-6cae373607effdd513024f7145a722f3R18

for #2, it looks like `replicationAttrs` should also have a property called `isSecondary`, and so my hunch would be to use that value explicitly, however upon testing this in a DR secondary i noticed that the actual value of `isSecondary` might be incorrect? i'm not sure why this would be -- i got this value by inspecting the cluster model, then console logging the `replicationAttrs`. mind taking a look into that?

![image](https://user-images.githubusercontent.com/903288/83197237-071b9780-a0f2-11ea-949f-6b875c8af49e.png)

Reply:
(also lol at those links above linking to some of the first vault PRs!!!)

Reply:
So, it turns out I didn't have to pass it `clusterMode`, we have the same cluster properties passed down from DR secondary as the others! 🎉 Thanks for the 🦅👁️ on that!

As far as using `isSecondary` explicitly, if there are only two modes it doesn't really matter which one we check for and which is the `else`... but, if there is another mode I'm not aware of (or might be in the future!) that would be helpful information. I'm going to go ahead and close out this comment since the code itself is outdated, but thanks again for the thorough feedback!